src/plugins/plugin-connect.ts

```diff
@@ -13,26 +13,50 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-'use strict';
 
-var urlParse = require('url').parse;
+// tslint:disable-next-line:no-reference
+/// <reference path="../types.d.ts" />
 
-var SUPPORTED_VERSIONS = '3.x';
+import {IncomingMessage, ServerResponse} from 'http';
+import * as shimmer from 'shimmer';
+import {parse as urlParse} from 'url';
 
-function createMiddleware(api) {
-  return function middleware(req, res, next) {
-    var options = {
-      name: urlParse(req.originalUrl).pathname,
+import {PluginTypes} from '..';
+
+import {connect_3} from './types';
+
+type Connect3 = typeof connect_3;
+// Connect docs note that routed requests have an originalUrl property.
+// https://github.com/senchalabs/connect/tree/3.6.5#appuseroute-fn
+type Request = IncomingMessage&{originalUrl?: string};
+
+const SUPPORTED_VERSIONS = '3.x';
+
+function getFirstHeader(req: IncomingMessage, key: string) {
```

Comment:
Maybe specify the return type as `string`?

Reply:
Actually it would have to be `string | null` since it might not exist. (Small detail -- choosing `| null` over `| undefined` since I don't want to pass around undefined values)